### PR TITLE
Fixed:Sales rule condtions are duplicated when data validation error …

### PR DESCRIPTION
**Pull request by mahesh-rajawat**
…occurs.

Fixed issue #23137 in this PR
### Description (*)

Fixed: Sales rule condtions are duplicated when data validation error occurs.
### Fixed Issues (if relevant)

1. #23137 

### Manual testing scenarios (*)

1. Create new sales rule and save it.
2. Edit this sales rule and try to through custom error from `Magento\Rule\Model\AbstractModel::validateData() function.`
3. Before this fix you will see in the conditions tab all are duplicates. https://prnt.sc/pm7fyt 
4. Fixed in this PR https://prnt.sc/pm7gks

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rodrigowebjump** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, here is your [new Magento instance](https://pr-25191.instances.magento-community.engineering).
Admin access: [https://pr-25191.instances.magento-community.engineering/admin](https://pr-25191.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`

File: app/code/Magento/SalesRule/Controller/Adminhtml/Promo/Quote/Edit.php [hunk 1]
```diff
@@ -57,6 +59,12 @@ public function execute()
                 $this->_redirect('sales_rule/*');
                 return;
             }
+
+            // set entered data if was error when we do save
+            $data = $this->_objectManager->get(\Magento\Backend\Model\Session::class)->getPageData(true);
```
Comment (aleron75) [inline app/code/Magento/SalesRule/Controller/Adminhtml/Promo/Quote/Edit.php]: If I'm not wrong, the `\Magento\Backend\Model\Session` is already in the `\Magento\Framework\App\Action\Context` which is an injected dependency, so there is no need to instantiate it through the object manager, what do you think?

Comment (mahesh-rajawat) [reply app/code/Magento/SalesRule/Controller/Adminhtml/Promo/Quote/Edit.php]: @aleron75 yes we can remove the objectManager here. i just moved the block to resolve this issue .

**Review by aleron75 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-6163](https://jira.corp.magento.com/browse/ENGCOM-6163) has been created to process this Pull Request
:eight_spoked_asterisk: @aleron75, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
Hi @maheshWebkul721. 
Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. Please cover your fix by automated tests.
Thanks!

**Comment by mahesh-rajawat** [comment]
@engcom-Alfa , i think all test has been covered, isn't it? No test seems to failed.

**Comment by engcom-Alfa** [comment]
@maheshWebkul721 You should cover your fix by automated tests with the scenario which leads to an issue. Tests should fail on the mainline and pass with your fix.
Thanks!

**Review by rodrigowebjump — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rodrigowebjump, thank you for the review.
[ENGCOM-6163](https://jira.corp.magento.com/browse/ENGCOM-6163) has been created to process this Pull Request
:eight_spoked_asterisk: @rodrigowebjump, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @maheshWebkul721. Thanks for collaboration. Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. Please cover your fix by automated tests.

**Comment by aleron75** [comment]
Hello @maheshWebkul721 just to add some more context to @VladimirZaets's request, for this specific case, you should write the necessary test scenarios that lead to the issue. 
Tests should fail on the mainline and pass with your fix.

To answer the question "which kind of tests should I write", the answer is:
pick the most lightweight (execution time-wise) test type that will provide sufficient coverage.

I hope this helps.

Thank you again!

**Comment by VladimirZaets** [comment]
@maheshWebkul721, I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Pull request closed without merging**